BrainPortal/app/models/tool_config.rb

```diff
@@ -593,15 +593,21 @@ def self.register_descriptor(descriptor, tool_name, tool_version) #:nodoc:
   def self.registered_boutiques_descriptor(tool_name, tool_version) #:nodoc:
     @_descriptors_ ||= {}
     key = [ tool_name, tool_version ] # two strings
+    @_descriptors_[key] = @_descriptors_[key]&.reload_if_file_timestamp_changed
     @_descriptors_[key]
   end
 
   def boutiques_descriptor
+    if @_descriptor_
```

Comment:
These 6 lines (601 to 606) should be at the same place you removed the line 604 in the original code. Otherwise you'te changing the priority logic